client_test.go

```diff
@@ -70,6 +69,7 @@ func TestMain(m *testing.M) {
 	if testing.Short() {
 		os.Exit(m.Run())
 	}
+	flag.Parse()
```

Comment:
The flag variables are used before `testContext` is called and in some cases when `testContext` is not called at all. Is this Parse even necessary still? Is it possible to do this change in master in a way that works with 1.12 and 1.13.